File: Searching/JavaCodes/Leetcode153.java
class Solution {
    
    public static int findMin(int[] nums) {
        int start=0,end= nums.length-1;

        while(start<end){
            int mid=start+(end-start+1)/2;
            if(nums[start]>nums[mid]){
                end=mid-1;
            }
            else{
                start=mid;
            }
        }
        int findIndex=start;
        int minIndex=(findIndex+1)%nums.length;
        return nums[minIndex];
        
    }
    
}

File: LogicOverSyntaxCodes/Searching/PythonCodes/Leetcode153.py
class Solution(object):
    def findMin(self, nums):
        start, end = 0, len(nums) - 1

        while start < end:
            mid = start + (end - start + 1) // 2
            if nums[start] > nums[mid]:
                end = mid - 1
            else:
                start = mid

        find_index = start
        min_index = (find_index + 1) % len(nums)
        return nums[min_index]
